wordpress-mobile/WordPress-iOS
# Issues/7675 refactor loginfields

Refs #7675
This does a bit of refactor to the LoginFields class to clarify its usage in certain scenarios and better organize some of its fields.  Doing this in advance of adding some new functionality to support social login.

To test:
Test the following auth flows:
- wpcom + magic link
- wpcom + password
- self-hosted
- saved safari credentials (iOS 10 only)
- 1password
- Sign up

Confirm that all still succeed. Confirm the epilogue still shows shows site info when its available.
Run the test suite. Confirm that NUX related tests pass.  In particular the ones for SigninHelpers and the LoginFacade.

Needs review: @nheagy
